### ISSUE TITLE
incrementFileName incrementing incorrect section of file name

### ISSUE THREAD

**Issue by xelvis42** [microsoft/vscode#54728]
Issue Type: <b>Bug</b>

I have a file named F1000-v117.au3. When copying/pasting the file, it creates a file named F1001-v117.au3. This is not the expected result. F1000 is the name of the file and not a version number that should be incremented. 

Changing the regex within the incrementFileName function fixes the issue. Basically, make sure the first section of the file name is numeric only. 

```
export function incrementFileName(name: string, isFolder: boolean): string {

	[other code...]

// 1.file.txt=>2.file.txt
    let prefixFileRegex = RegExp('(\\d+)(' + separators + '.*)(\\..*)$');
    if (!isFolder && name.match(prefixFileRegex)) {
      return name.replace(prefixFileRegex, (match, g1?, g2?, g3?) => {
        let number = parseInt(g1);
        return number < maxNumber
          ? this.pad(number + 1, g1.length) + g2 + g3
          : this.format('{0}{1}.1{2}', g1, g2, g3);
      });
    }
        [other code...]
}
```
to 
```
export function incrementFileName(name: string, isFolder: boolean): string {

	[other code...]

// 1.file.txt=>2.file.txt
    let prefixFileRegex = RegExp('(\\d$)(' + separators + '.*)(\\..*)$');
    if (!isFolder && name.match(prefixFileRegex)) {
      return name.replace(prefixFileRegex, (match, g1?, g2?, g3?) => {
        let number = parseInt(g1);
        return number < maxNumber
          ? this.pad(number + 1, g1.length) + g2 + g3
          : this.format('{0}{1}.1{2}', g1, g2, g3);
      });
    }
        [other code...]
}
```

VS Code version: Code 1.25.1 (1dfc5e557209371715f655691b1235b6b26a06be, 2018-07-11T15:43:11.471Z)
OS version: Windows_NT ia32 6.1.7601

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4800MQ CPU @ 2.70GHz (8 x 2693)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|2.92GB (0.12GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe|
|Screen Reader|yes|
|VM|0%|

</details>

**Comment by isidorn** [comment]
PR for this is welcome
It would be great if the PR contains a unit test https://github.com/Microsoft/vscode/blob/1c14e736d279219324f11ddbe1f55e776ca89d26/src/vs/workbench/parts/files/test/electron-browser/fileActions.test.ts#L15

@Krzysztof-Cieslak fyi

**Comment by Krzysztof-Cieslak** [comment]
I'll take a look tomorrow.

**Comment by roblourens** [comment]
`F1000-v117.1.au3` is the expected result right?

**Comment by roblourens** [comment]
Actually it seems like other cases are broken now.

- Have a file `test1.txt`, copy and paste it
- The new file is `test1.1.txt`, in 1.26 it was `test2.txt`

**Comment by isidorn** [comment]
Thanks for finding that @roblourens 
Due to that I had to revert the original fix. So now copying test1.txt gives test2.txt
However this issue is now not fixed, but it was like that in previous stable so nothing urgent -> removing mielstone

**Comment by helsont** [comment]
Hey guys, I took a crack at fixing this regex have some questions on how the naming / number schema works and I want to make sure I have an understanding of the naming schema.

We have our cases here:

suffix:
```
file.1.txt=>file.2.txt
```

prefix:
```
1.file.txt=>2.file.txt
```
single number:
```
1.txt=>2.txt
```
no number:
```
file.txt=>file.1.txt
```
ends with number:
```
folder.1=>folder.2
```
mixed (new case / why the above PR broke):
```
name1.txt=>name1.1.txt
name1.test.txt=>name1.test.1.txt (???)
```
prefixed with number:
```
1.folder=>2.folder
```

The issue is that mixing letters and numbers triggers the prefix regex which incorrectly increments the number. But all the different variations of prefix / suffix and single number / mixed with letters make this particularly tricky. 

It seems like edits need to be made to existing regexes to ensure that they are matching digit only groups. However, the above PR regex edits and the edits I've made consistently break other regexes. 

Any ideas or considerations here?
